api/v1alpha3/types.go

```diff
@@ -137,9 +142,14 @@ type IngressRules []*IngressRule
 // PublicIP defines an Azure public IP address.
 type PublicIP struct {
 	ID        string `json:"id,omitempty"`
-	Name      string `json:"name,omitempty"`
 	IPAddress string `json:"ipAddress,omitempty"`
 	DNSName   string `json:"dnsName,omitempty"`
+	Name      string `json:"name,omitempty"`
+}
+
+// PublicIPSpec defines the inputs to create an Azure public IP address.
+type PublicIPSpec struct {
+	Name string `json:"name,omitempty"`
```

Comment:
do we really need this new type? Seems like it's a subset of the above `PublicIP` type and since all the fields besides Name are optional we should be able to reuse it? Is it because this is only meant to provide the name of an already existing IP?

Reply:
also what about resource group? Does this imply same RG as the rest of the cluster? BYO vnet allows a different RG for the vnet

Reply:
existing public IP is used only for status and exposes fields like IPAddress which wouldn't actually be settable from spec. I'm fine with adding a validating webhook against populating some of those? or else modifying public IP itself to look more like a k8s object: 

```go
// PublicIP defines an Azure public IP address.
type PublicIP struct {
	Spec
        Status
}

// PublicIPSpec defines the inputs to create an Azure public IP address.
type PublicIPSpec struct {
	DNSName   string `json:"dnsName,omitempty"`
	Name string `json:"name,omitempty"`
}

// PublicIPStatus defines the outputs from creating an Azure public IP address.
type PublicIPStatus struct {
	ID        string `json:"id,omitempty"`
	IPAddress string `json:"ipAddress,omitempty"`
}
```

that seems a little heavy though. Reusing the type + validating webhook seems like a nice compromise?

Reply:
Using a separate type has the benefit of safety on validating the struct -- when we add new fields, we would have to exclude any status-only fields in the future. That seems prone to error. So maybe splitting to spec/status is better? But I don't think DNS name would even be custom right now

Reply:
lmk what you think here, after fiddling with unit tests this is the main action item I see remaining

Reply:
@devigned do you have any thoughts on the above?

Reply:
I don't feel strongly about this, but I do prefer the proposal with `PublicIP` being composed of two structures. It's a bit more verbose, but it is clear in the separation.